src/App.jsx

```diff
@@ -1,20 +1,81 @@
 import './App.css';
 import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useCallback } from 'react';
 
-const useSemiPersistentState = () => {
-  const [value, setValue] = useState(JSON.parse(localStorage.getItem('savedTodoList')) || []);
-  useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(value));
-  }, [value]);
-  return [value, setValue];
-};
+const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
 
 const App = () => {
-  const [todoList, setTodoList] = useSemiPersistentState();
-  const addTodo = (newTodo) => {
-    setTodoList([newTodo, ...todoList])
+
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isAdding, setIsAdding] = useState(false);
+
+  // useCallBack for pessimistic rendering
+  const callbackFetchData = useCallback(async () => {
```

Comment:
See my comment here - https://github.com/SergeDot/react-todo/pull/7/files#r1569581634

Reply:
Changed per the recommendations:
1. removed the useCallBack and fetching data too many times
2. added error handler and error messages
3. replaced fetch with axios
4. modified empty string checker
5. minor refactoring